src/hass.tsx

```diff
@@ -239,14 +248,20 @@ export default class HomeAssistant {
         if (!promiseHandler) {
           console.warn("[hass] No pending request found for event", payload.id);
         } else {
-          let [resolve, reject] = promiseHandler;
+          let [resolve, reject, span] = promiseHandler;
           this._pendingRequests.delete(payload.id);
           if (payload.success) {
+            if (span) {
+              span.setStatus(SpanStatus.Ok);
+            }
             setTimeout(() => {
               this._pendingChanges.delete(payload.id);
             }, 1000);
             resolve(payload);
           } else if (payload.error) {
+            if (span) {
+              span.setStatus(SpanStatus.UnknownError);
```

Comment:
understanding how to map other systems status codes to our status codes will be a challenge for folks.. going to try an exercise of doing this w/ hass